src/Voron/Data/Tables/TableSchema.IndexDefinitions.cs

```diff
@@ -362,8 +376,25 @@ public static DynamicKeyIndexDef ReadFrom(ByteStringContext context, ref TableVa
                 // read if this index has OnEntryChanged
                 currentPtr = input.Read(5, out currentSize);
                 var hasIndexEntryChangedDelegate = Convert.ToBoolean(*currentPtr);
-                if (hasIndexEntryChangedDelegate == false) 
-                    return indexDef;
+                if (hasIndexEntryChangedDelegate == false)
+                {
+                    if (input.Count == 6)
```

Comment:
Is is for backward compatibility purposes? If yes, then please add a comment about that

Reply:
done